samples/js/splore/src/main.js

```diff
@@ -222,30 +248,38 @@ function handleNodeClick(e: MouseEvent, id: string) {
   }
 }
 
-class Prompt extends React.Component<void, {}, void> {
+type PromptProps = {
+  msg: string,
+}
+
+class Prompt extends React.Component<void, PromptProps, void> {
   render(): React.Element<any> {
     const fontStyle: {[key: string]: any} = {
       fontFamily: 'Menlo',
       fontSize: 14,
     };
     const inputStyle = Object.assign(fontStyle, {}, {width: '50ex', marginBottom: '0.5em'});
+    const demoServer = 'https://demo.noms.io/cli-tour';
+
     return <div style={{display: 'flex', height: '100%', alignItems: 'center',
       justifyContent: 'center'}}>
       <div style={fontStyle}>
-        Can haz database?
+        {this.props.msg}
         <form style={{margin:'0.5em 0'}} onSubmit={e => this._handleOnSubmit(e)}>
           <input type='text' ref='db' autoFocus={true} style={inputStyle}
-            defaultValue={params.db || 'https://demo.noms.io/cli-tour'}
-            placeholder='noms database URL'
+            defaultValue={params.db}
+            placeholder={`database URL (${demoServer})`}
+          />
+          or
+          <input type='text' ref='p' style={inputStyle}
```

Comment:
Not 100% sure what you mean. As in require `db`, then if there is `p` interpret `db+p` as a path spec (otherwise just `db` as a db spec)?

That would work. The nice thing about having it being a full spec is that it's consistent with our samples. We don't say `noms show --db http://demo.noms.io/cli-tour --p sf-crime.value[42]`, we say `noms show http://demo.noms.io::sf-crime.value[42]`. I like that there is a parallel between the two, IMO we should be using paths when we have the opportunity.

On the other hand it does make the code simpler, and the UI a little less ambiguous.